drivers/vl53l1x/vl53l1x.c

```diff
@@ -0,0 +1,1292 @@
+    #endif
+
+    /* save the measurement period */
+    dev->params.period = period_ms;
+
+    return VL53L1X_OK;
+}
+
+int vl53l1x_set_distance_mode(vl53l1x_t *dev, vl53l1x_dist_mode_t mode)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("mode=%u", dev, mode);
+
+    #if MODULE_VL53L1X_ST_API
+
+    EXEC_RET(VL53L1_SetDistanceMode((VL53L1_Dev_t*)dev, mode));
+
+    #else
+    /* Copyrighted POLOLU / ST API code begin */
+
+    /* save existing timing budget */
+    uint32_t budget_us;
+    EXEC_RET(vl53l1x_get_timing_budget(dev, &budget_us));
+
+    switch (mode) {
+        case VL53L1X_DIST_SHORT:
+            /* from VL53L1_preset_mode_standard_ranging_short_range() */
+            /* timing config */
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_A, 0x07));
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_B, 0x05));
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VALID_PHASE_HIGH, 0x38));
+
+            /* dynamic config */
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD0, 0x07));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD1, 0x05));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD0, 6));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD1, 6));
+            break;
+
+        case VL53L1X_DIST_MEDIUM:
+            /* from VL53L1_preset_mode_standard_ranging() */
+            /* timing config */
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_A, 0x0b));
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_B, 0x09));
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VALID_PHASE_HIGH, 0x78));
+
+            /* dynamic config */
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD0, 0x0b));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD1, 0x09));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD0, 10));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD1, 10));
+            break;
+
+        case VL53L1X_DIST_LONG:
+            /* from VL53L1_preset_mode_standard_ranging_long_range() */
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_A, 0x0f));
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_B, 0x0d));
+            EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VALID_PHASE_HIGH, 0xb8));
+
+            /* dynamic config */
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD0, 0x0f));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD1, 0x0d));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD0, 14));
+            EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD1, 14));
+            break;
+
+        default:
+            /* unrecognized mode - do nothing */
+            return VL53L1X_ERROR_INV_ARG;
+    }
```

Comment:
```suggestion
    switch (mode) {
    case VL53L1X_DIST_SHORT:
        /* from VL53L1_preset_mode_standard_ranging_short_range() */
        /* timing config */
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_A, 0x07));
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_B, 0x05));
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VALID_PHASE_HIGH, 0x38));

        /* dynamic config */
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD0, 0x07));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD1, 0x05));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD0, 6));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD1, 6));
        break;

    case VL53L1X_DIST_MEDIUM:
        /* from VL53L1_preset_mode_standard_ranging() */
        /* timing config */
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_A, 0x0b));
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_B, 0x09));
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VALID_PHASE_HIGH, 0x78));

        /* dynamic config */
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD0, 0x0b));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD1, 0x09));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD0, 10));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD1, 10));
        break;

    case VL53L1X_DIST_LONG:
        /* from VL53L1_preset_mode_standard_ranging_long_range() */
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_A, 0x0f));
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VCSEL_PERIOD_B, 0x0d));
        EXEC_RET(_write_byte(dev, VL53L1X_RANGE_CONFIG__VALID_PHASE_HIGH, 0xb8));

        /* dynamic config */
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD0, 0x0f));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__WOI_SD1, 0x0d));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD0, 14));
        EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__INITIAL_PHASE_SD1, 14));
        break;

    default:
        /* unrecognized mode - do nothing */
        return VL53L1X_ERROR_INV_ARG;
    }
```

Reply:
Is this change really necessary? I really don't like this style because it reduces the readability.

I know, according to the Linux coding style it is the preferred way, but does is it mean that it is forbidden to use double Indentation? In Linux style 8 chars are used for indentation and the preferred switch/case indentation is a way to reduce the indentation levels to avoid that the code moves too far to the right on a 80 char terminal. 

It might be considered as a violation of their own "idea behind indentation to clearly define where a block of control starts and ends". The switch statement starts a block and all statements inside a block would have to be indented including the case statements.

I know, we could discuss about it for hours. Finally, when I'm crawling through RIOT code, both indentation schemes are used for switch/case statements, also in mature code. Therefore, I would prefer to leave it as it is. Would that be OK?

Reply:
Fixed with commit https://github.com/RIOT-OS/RIOT/pull/10363/commits/71fd9adb7b92b9a63b3b84f731b7cdec408065c3.